frontend2/src/views/Login.tsx

```diff
@@ -32,13 +35,15 @@ const Login: React.FC = () => {
 
   const onSubmit: SubmitHandler<LoginFormInput> = async (data) => {
     try {
-      setLoginError(undefined);
-      await apiLogin(data.username, data.password);
-      login(await getUserUserProfile());
-    } catch (error) {
-      setLoginError(
-        "Error logging in. Did you enter your username and password correctly?",
-      );
+      await toast.promise(login(data.username, data.password, queryClient), {
+        loading: "Logging in...",
+        success: "Logged in!",
+        error:
+          "Error logging in. Did you enter your username and password correctly?",
+      });
+      navigate(episodeId !== undefined ? `/${episodeId}/home` : "/");
```

Comment:
I don't think we need this extra navigate (see the useEffect above that should accomplish the same thing)

Reply:
💯